### PR TITLE
Makes getting date formatters more robust

### PR DESCRIPTION
**Pull request by cschierle**
Fixes: SIRI-525

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/kernel/nls/NLS.java [hunk 1]
```diff
@@ -667,10 +663,16 @@ public static DateTimeFormatter getDateTimeFormat(String lang) {
...
-                                                          "NLS.patternDateTime.withoutSeconds",
-                                                          l)));
+        return safeGetDateTimeFormat(lang, dateTimeFormatters, "NLS.patternDateTime.withoutSeconds");
+    }
+
+    private static DateTimeFormatter safeGetDateTimeFormat(String lang,
+                                                           Map<String, DateTimeFormatter> formatters,
+                                                           String nlsProperty) {
+        if (Strings.isEmpty(lang)) {
+            lang = getCurrentLang();
```
Comment (andyHa) [inline src/main/java/sirius/kernel/nls/NLS.java:673]: assigning to parameters is commonly not the preferred choice - we could just return here...

Comment (andyHa) [inline src/main/java/sirius/kernel/nls/NLS.java:673]:
```suggestion
             return formatters.computeIfAbsent(getCurrentLang(), language -> DateTimeFormatter.ofPattern(get(nlsProperty, language)));
```

File: src/main/java/sirius/kernel/nls/NLS.java [hunk 2]
```diff
@@ -667,10 +663,16 @@ public static DateTimeFormatter getDateTimeFormat(String lang) {
...
+        }
+        return formatters.computeIfAbsent(lang, l -> DateTimeFormatter.ofPattern(get(nlsProperty, l)));
```
Comment (andyHa) [inline src/main/java/sirius/kernel/nls/NLS.java:675]:
```suggestion
        return formatters.computeIfAbsent(lang, ignored -> DateTimeFormatter.ofPattern(get(nlsProperty, lang)));
```

File: src/main/java/sirius/kernel/commons/Value.java [hunk 3]
```diff
@@ -1329,6 +1329,9 @@ public Value tryParseLocalDate(DateTimeFormatter formatter) {
      * @return the parsed LocalDate wrapped as a Value, or the Value itself if it was empty or a parsing error occurred
      */
     public Value tryParseLocalDate(String languageCode) {
+        if (Strings.isEmpty(languageCode)) {
```
Comment (andyHa) [inline src/main/java/sirius/kernel/commons/Value.java:1332]: in every other location, we'd use getCurrentLang() here ?! is this consistent?

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/kernel/commons/Value.java (hunk 3 above)
Comment (cschierle) [reply src/main/java/sirius/kernel/commons/Value.java:1332]: This method only **tries** to parse the value as a LocalDate. So when an empty language is passed, there is no need to try any further.

I think for a `tryParseLocalDate` returning without defaulting to the current language is the expected behavior. If this was a `parseLocalDate`, then the default should be used.

**Review by fhaScireum — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

File: src/main/java/sirius/kernel/commons/Value.java (hunk 3 above)
Comment (andyHa) [reply src/main/java/sirius/kernel/commons/Value.java:1332]: Still in all other methods "null" is used to represent the default language. And the try is meant to "tryParse" not to check parameters(?). I'd say if parseLocalDate yields a value, tryParseLocalDate with **the same parameters** should also return a value.

**Review by andyHa — APPROVED** [review]

**Pull request merged**